src/build_system.ml

```diff
@@ -815,3 +817,6 @@ let build_rules t ?(recursive=false) targets =
     die "dependency cycle detected:\n   %s"
       (List.map cycle ~f:(fun rule -> Path.to_string (Pset.choose rule.Rule.targets))
        |> String.concat ~sep:"\n-> ")
+
+let dump_trace t =
```

Comment:
In which case I'd have to expose the `Trace` module and its `dump` function and add an accessor for the the trace file of `Build_system.t`.

Should I do that?